functorch/csrc/BatchingRegistrations.cpp

```diff
@@ -557,6 +557,32 @@ Tensor cat_batching_rule(TensorList tensors, int64_t dim) {
   return physical_views[0].getPhysicalToLogicalMap().apply(result);
 }
 
+Tensor block_diag_batching_rule(TensorList tensors) {
```

Comment:
(no action required) For some more context... BatchingRegistrations.cpp is for all the legacy batching rules and is "deprecated", but since we haven't actually gotten TensorList inputs to work with the new batching rule API, this is indeed our only option for now. The downside is that the legacy API is a bit difficult to work with as you've probably discovered in here